src/objective/regression_objective.hpp

```diff
@@ -51,6 +51,60 @@ class RegressionL2loss: public ObjectiveFunction {
   const float* weights_;
 };
 
+
+class RegressionL1loss: public ObjectiveFunction {
+public:
+  explicit RegressionL1loss(const ObjectiveConfig& config) {}
+
+  ~RegressionL1loss() {}
+
+  void Init(const Metadata& metadata, data_size_t num_data) override {
+    num_data_ = num_data;
+    label_ = metadata.label();
+    weights_ = metadata.weights();
+  }
+
+  void GetGradients(const score_t* score, score_t* gradients,
+                    score_t* hessians) const override {
+    if (weights_ == nullptr) {
+      #pragma omp parallel for schedule(static)
+      for (data_size_t i = 0; i < num_data_; ++i) {
+        const double diff = score[i] - label_[i];
+        if (diff >= 0.0) {
+          gradients[i] = 1.0;
+        } else {
+          gradients[i] = -1.0;
+        }
+        hessians[i] = 0.0;
+      }
+    } else {
+      #pragma omp parallel for schedule(static)
+      for (data_size_t i = 0; i < num_data_; ++i) {
+        const double diff = score[i] - label_[i];
+        if (diff >= 0.0) {
+          gradients[i] = weights_[i];
+        } else {
+          gradients[i] = -weights_[i];
+        }
+        hessians[i] = 0.0;
```

Comment:
A friend of mine who majors in theoretical physics tells me that it may be good to approximate hessians with Gaussian function.
Gaussian function is often used to approximate [ultrashort pulse](https://en.wikipedia.org/wiki/Ultrashort_pulse "Ultrashort pulse - Wikipedia").

Which do you think it is better to use `all hessians == 1` or Gaussian function.